packages/@expo/cli/src/start/server/metro/inspector-proxy/proxy.ts

```diff
@@ -0,0 +1,167 @@
+import type { Server as HttpServer, IncomingMessage, ServerResponse } from 'http';
+import type { Server as HttpsServer } from 'https';
+import type { InspectorProxy as MetroProxy, Device as MetroDevice } from 'metro-inspector-proxy';
+import type { AddressInfo } from 'net';
+import { parse } from 'url';
+import type { Server as WSServer } from 'ws';
+
+import { importMetroInspectorWebSocketFromProject } from '../resolveFromProject';
+
+const WS_DEVICE_URL = '/inspector/device';
+const WS_DEBUGGER_URL = '/inspector/debug';
+const WS_GENERIC_ERROR_STATUS = 1011;
+
+const debug = require('debug')('expo:metro:inspector-proxy:proxy') as typeof console.log;
+
+// This is a workaround for `ConstructorType` not working on dynamically generated classes
+type Instantiatable<Instance> = new (...args: any) => Instance;
+
+export class ExpoInspectorProxy<D extends MetroDevice> {
+  constructor(
+    public readonly metroProxy: MetroProxy,
+    private DeviceClass: Instantiatable<D>,
+    public readonly devices: Map<number, D> = new Map()
+  ) {
+    this.metroProxy = metroProxy;
+    // monkey-patch the device list to expose it within the metro inspector
+    this.metroProxy._devices = this.devices;
+
+    // force httpEndpointMiddleware to be bound to this proxy instance
+    this.httpEndpointMiddleware = this.httpEndpointMiddleware.bind(this);
+  }
+
+  /**
+   * When the Metro is fully initialized, resolve the server address being used
+   * This is required to properly reference sourcemaps for the debugger.
+   */
+  public onServerReady(server: HttpServer | HttpsServer) {
+    const address = server.address();
+
+    if (typeof address === 'string') {
+      throw new Error(`Inspector proxy could not resolve the server address, got ${address}`);
+    } else if (address === null) {
+      throw new Error(`Inspector proxy could not resolve the server address, got "null"`);
+    }
+
+    this.setServerAddress(address);
+  }
+
+  /** @see https://chromedevtools.github.io/devtools-protocol/#endpoints */
+  public httpEndpointMiddleware(
+    req: IncomingMessage,
+    res: ServerResponse,
+    next: (error?: Error) => any
+  ) {
+    return this.metroProxy.processRequest(req, res, next);
+  }
+
+  public createWebSocketEndpoints(port = 1111): Record<string, WSServer> {
+    // Auto-initialize the server address on a best-guess basis
+    // If `onServerReady` is not called, this might break sourcemap references
+    this.setServerAddress({ port, family: 'IPv4', address: '' });
+
+    return {
+      [WS_DEVICE_URL]: this.createDeviceWebSocketServer(),
+      [WS_DEBUGGER_URL]: this.createDebuggerWebSocketServer(),
+    };
+  }
+
+  private setServerAddress({ port, family, address }: AddressInfo) {
+    if (family === 'IPv6') {
+      this.metroProxy._serverAddressWithPort = `${address ?? '[::1]'}:${port}`;
```

Comment:
```suggestion
      this.metroProxy._serverAddressWithPort = `[${address ?? '::1'}]:${port}`;
```
the ipv6 address does include the brackets. my `server.address()` just returns `::1`.

Reply:
Oh man, gotta love IPv6. Thanks, will fix!